### ISSUE TITLE
XrdCl :: handle for configurables customization (timeouts/ XRD_ configurables) per copy job

### ISSUE THREAD

**Issue by adriansev** [xrootd/xrootd#1696]
Hi! It would be useful if there would be a handle (available also to python bindings)
to customize each copy job configurables. This could be useful to add some kind of adaptations
of the job depending on the importance and the content of the file.

**Comment by amadio** [comment]
Actually, this one is not the duplicate, it should be the new one. Let's keep this one open and close the other as duplicate.

**Comment by adriansev** [comment]
Hi @amadio! So, will the timeouts in the arguments of add_job be private to that job?
Can i use different values and expect that each job to have it's own values? 

So, most likely i'm wrong as i lack overview understanding of xrootd code, but i was looking at 
https://github.com/xrootd/xrootd/blob/master/bindings/python/src/PyXRootDCopyProcess.cc#L123-L124

and instead of:
```
    XrdCl::PropertyList properties;
    self->results->push_back(XrdCl::PropertyList());

    properties.Set( "source",          source          );
	.... bunch of sets ....
```

i would have expected something like:
```
    XrdCl::PropertyList properties;
    properties.Set( "source",          source          );
	.... bunch of sets ....

    self->results->push_back(properties);
```

Is this a false problem?
Thanks a lot!

**Comment by amadio** [comment]
After the lines you linked to, there is

https://github.com/xrootd/xrootd/blob/b7cc4cd2c2b94bf193b57096c81464743ad3838e/bindings/python/src/PyXRootDCopyProcess.cc#L159-L160

**Comment by adriansev** [comment]
oh, i see! then i get that each copy job have it's own list of settings, namely https://github.com/xrootd/xrootd/blob/master/bindings/python/src/PyXRootDCopyProcess.cc#L126-L145 
and `CpRetry` and `CpRetryPolicy` are global ?

**Comment by amadio** [comment]
Yes, correct. I think that we could adjust the retry settings, though. That looks odd that only these two are put in the environment like that. In any case, please give it a try and in case you need any changes for making what you need work, please let us know here.
